### PR TITLE
feat: Add an option to enable/disable email collect box

### LINKED ISSUE #2040: Enhancement: Inbox setting to disable email collect box in widget
- Way to disable/customise the email collect message "Give them a way to reach"

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes #2040

### CODE REVIEW

File: db/migrate/20210607164125_add_email_collect_to_inboxes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddEmailCollectToInboxes < ActiveRecord::Migration[6.0]
+  def change
+    add_column :inboxes, :enable_email_collect, :boolean
```
Comment (pranavrajs) [inline db/migrate/20210607164125_add_email_collect_to_inboxes.rb:3]: Please add a default to true.

File: app/services/message_templates/hook_execution_service.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def perform
     # ::MessageTemplates::Template::OutOfOffice.new(conversation: conversation).perform if should_send_out_of_office_message?
 
     ::MessageTemplates::Template::Greeting.new(conversation: conversation).perform if should_send_greeting?
-    ::MessageTemplates::Template::EmailCollect.new(conversation: conversation).perform if should_send_email_collect?
+    ::MessageTemplates::Template::EmailCollect.new(conversation: conversation).perform if inbox.enable_email_collect && should_send_email_collect?
```
Comment (pranavrajs) [inline app/services/message_templates/hook_execution_service.rb:12]: Please add a spec to cover this case.

File: db/schema.rb [hunk 3]
```diff
@@ -322,6 +322,7 @@
     t.boolean "working_hours_enabled", default: false
     t.string "out_of_office_message"
     t.string "timezone", default: "UTC"
+    t.boolean "enable_email_collect"
```
Comment (sojan-official) [inline db/schema.rb:325]: seems like the `default: true` isn't captured in the schema. Let's rerun the migration and ensure that is captured.

**Review by sojan-official — DISMISSED** [review]
ruby code LGTM

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**